### ISSUE TITLE
Weather data sources

### ISSUE THREAD

**Issue by HonestAbe7** [davidtakac/prognoza#12]
Please consider adding more weather data sources like openweather.org, etc . Thanks

**Comment by davidtakac** [comment]
Unfortunately, OpenWeatherMap [is not free](https://gitlab.com/fdroid/fdroiddata/-/issues/1439), so I will not be adding support for it. 

I am open to other weather providers as long as they are free, though.

**Comment by JoshuaKimsey** [comment]
You could include data from the United States' National Weather Service. Any data released by them is automatically in the public domain according to federal law.

**Comment by davidtakac** [comment]
That's great! I will check it out, thank you.

**Comment by HonestAbe7** [comment]
That would be cool if you could include the NWS because they do give out severe weather alerts.

**Comment by 0-5-0** [comment]
Environment Canada also makes their weather data public for free, all they ask is that you credit Environment and Climate Change Canada like you do for MET Norway.

Here's another repo that might be useful for comparison or to see how the data might be implemented in Android. https://github.com/austinhuang0131/CanadaWeather or https://f-droid.org/en/packages/me.austinhuang.caweather/

Also found this: https://github.com/michaeldavie/env_canada

**Comment by davidtakac** [comment]
Thank you for the suggestion, but unfortunately, the Canada Weather app is also marked with non-free net :/

**Comment by 0-5-0** [comment]
That seems odd, Environment Canada data is free but they do ask for just a simple acknowledgement: https://climate.weather.gc.ca/prods_servs/attachment1_e.html

This is a broader agreement for all Open Data provided by the Government of Canada with a similar statement: http://open.canada.ca/en/open-government-licence-canada

From F-Droid it says non-free net means in part: "This Anti-Feature is applied to apps that promote or depend entirely on a Non-Free network service which is impossible, or not easy to replace". It does seem like the majority of weather apps (at least all the ones I saw on Fdroid) have the same non-free network tag. Maybe because weather data is not easily replaceable.

This is another weather app on F-Droid but with free NOAA data and it's tagged non-free. https://f-droid.org/en/packages/joshuatee.wx/

F-Droid also says: "This antifeature would not apply, if there is a simple configuration option that allows pointing the app to a running instance of an alternative, publicly available, self-hostable, free software server solution."

I did notice Prognoza isn't searchable on Fdroid and that it doesn't seem to have the same tag on IzzyOnDroid (maybe Izzy is more relaxed).

**Comment by davidtakac** [comment]
MET Norway and Environment Canada seem to have the exact same license terms, and neither have the server source code available, yet one is considered free net and the other is not. Kind of confusing 🤔 

What's even more confusing is that [NWS Weather Alerts Widget](https://www.f-droid.org/packages/net.justdave.nwsweatheralertswidget/) also uses data from NWS but doesn't have the anti-feature. 

Either way, Prognoza could be exempt from this anti-feature because it wouldn't "entirely" depend on non-free services. The user can always choose MET Norway which is considered free. 

Although I'm not sure NWS and Environment Canada should be considered non-free in the first place! I'll research this some more and maybe open an issue on F-Droid to try and get some more info.

**Comment by codeinabox** [comment]
The Met Office in the UK also has weather data licenced under the Open Government Licence - is that appropriate?

https://www.metoffice.gov.uk/services/data/datapoint/terms-and-conditions---datapoint

**Comment by davidtakac** [comment]
I added support for Open-Meteo in version 3.3.0. This provider's great because: 

- It's free (freedom and beer)
- It automatically chooses the best underlying data source for the given coordinates. Some examples:
  - For San Francisco, it pulls from NWS (@JoshuaKimsey)
  - For Paris, it pulls from Meteofrance. 
  - For Toronto, it pulls from CMC (@0-5-0)
  - and so on, there's a few other services that are covered: https://open-meteo.com/en/docs

**Comment by davidtakac** [comment]
> The Met Office in the UK also has weather data licenced under the Open Government Licence - is that appropriate?

@codeinabox I checked out their API and it requires an API key, so support is not planned in the near future. Please try version 3.3.0 and switch to Open-Meteo, they may provide more accurate data for the UK.

**Comment by 0-5-0** [comment]
> I added support for Open-Meteo in version 3.3.0. ...

Nice find, I haven't seen it used or mentioned anywhere else for Android apps. Looks great.

I also opened a discussion on Open-Meteo's Github page to see if they might add national weather service alerts to their API: https://github.com/open-meteo/open-meteo/discussions/183

**Comment by ak-42** [comment]
@davidtakac Hi. Another source that you might consider implementing is https://pirateweather.net/ (despite the name, there's no piracy involved).

**Comment by davidtakac** [comment]
Hi! I'll check it out, first I heard of it. On first glance, looks to be similar to Open-Meteo :)

**Comment by rsssxd** [comment]
Open- Meteo supports a wide range of data such as UV, Humidity, Wind direction. Add support for users who choose Open- Meteo.

**Comment by davidtakac** [comment]
@rsssxd Yes, all these are planned. I even have humidity data ready in the current app, but nowhere to put it!

I'm working on a redesign for these features, as well as the inclusion of graphs. The current design would look too crowded with all this extra info, and I want it to look good, so it's taking longer 😄

**Comment by brian6932** [comment]
Thoughts on adding https://pirateweather.net/ ?
Demo site: https://merrysky.net/

**Comment by davidtakac** [comment]
I don't think I will add support for PirateWeather because: 
- It requires an API key
- It uses data from government agencies, which is already fulfilled by Open-Meteo

**Comment by Fqwe1** [comment]
Hi, add [brightsky.dev](https://brightsky.dev/). Thanks!
